engine/src/stack.cpp

```diff
@@ -101,6 +101,31 @@ std::string PluginConf::canonical() const {
 
 // --------------------------------------------------------------------------------------------- //
 
+SimulatorSchema::SimulatorSchema() {
+  this->set_transform_schema([](fable::schema::Struct&& s) -> fable::schema::Box {
+    s.set_property("name", id_prototype("globally unique identifier for simulator"));
+    return s;
+  });
+}
+
+ControllerSchema::ControllerSchema() {
+  this->set_transform_schema([](fable::schema::Struct&& s) -> fable::schema::Box {
+    s.set_property("name", id_prototype("globally unique identifier for controller"));
```

Comment:
Now, there is some redundancy between the factory schema transformation defined here and the schema definition in `ControllerConf` (s. [here](https://github.com/eclipse/cloe/blob/ben/fix-factory-schema/engine/src/stack.hpp#L549)). But I guess a solution is out of the scope of this PR

Reply:
That is correct, and is also out of the scope of this PR. One step at a time, in particular once the benefit is high enough.